api/src/main/java/org/apache/iceberg/SortOrder.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import org.apache.iceberg.SortField.Direction;
+import org.apache.iceberg.SortField.NullOrder;
+import org.apache.iceberg.SortTransforms.SortTransform;
+import org.apache.iceberg.types.Types;
+
+// TODO: toString()
+public class SortOrder implements Serializable {
+
+  private static final SortOrder UNSORTED_ORDER = new SortOrder(new Schema(), 0, ImmutableList.of());
```

Comment:
Does this mean that order ID 0 is reserved for the unsorted order?

Reply:
Nit: `Collections.emptyList()` also returns an immutable list